File: CS/NavigationDemo.Wpf/DemoTesting/Fixtures.cs
using System;
using System.Collections;
using System.Linq;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Controls.Primitives;
using System.Windows.Input;
using System.Windows.Media.Animation;
using DevExpress.Xpf.Core;
using DevExpress.Xpf.DemoBase.DemoTesting;
using DevExpress.Xpf.Editors;

namespace NavigationDemo.Tests {
    public class AccordionCheckAllDemosFixture : CheckAllDemosFixture {
        Type[] skipMemoryLeaksCheckModules = new Type[] { typeof(FileSearchDemoModule) };
        protected override bool CheckMemoryLeaks(Type moduleType) {
            return !skipMemoryLeaksCheckModules.Contains(moduleType);
        }
        protected override bool AllowSwitchToTheTheme(Type moduleType, Theme theme) {
            return theme != Theme.HybridApp;
        }
    }
    public class AccordionDemoModulesAccessor : DemoModulesAccessor<AccordionDemoModule> {
        public AccordionDemoModulesAccessor(BaseDemoTestingFixture fixture)
            : base(fixture) {
        }
    }
    public abstract class BaseAccordionDemoTestingFixture : BaseDemoTestingFixture {
        protected AccordionDemoModulesAccessor ModuleAccessor { get; private set; }
        public BaseAccordionDemoTestingFixture() {
            ModuleAccessor = GetModulesAccessor();
        }
        protected abstract AccordionDemoModulesAccessor GetModulesAccessor();
    }
}

File: VB/NavigationDemo.Wpf/DemoTesting/Fixtures.vb
Imports System
Imports System.Collections
Imports System.Linq
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Controls.Primitives
Imports System.Windows.Input
Imports System.Windows.Media.Animation
Imports DevExpress.Xpf.Core
Imports DevExpress.Xpf.DemoBase.DemoTesting
Imports DevExpress.Xpf.Editors

Namespace NavigationDemo.Tests
    Public Class AccordionCheckAllDemosFixture
        Inherits CheckAllDemosFixture

        Private skipMemoryLeaksCheckModules() As Type = { GetType(FileSearchDemoModule) }
        Protected Overrides Function CheckMemoryLeaks(ByVal moduleType As Type) As Boolean
            Return Not skipMemoryLeaksCheckModules.Contains(moduleType)
        End Function
        Protected Overrides Function AllowSwitchToTheTheme(ByVal moduleType As Type, ByVal theme As Theme) As Boolean
            Return theme IsNot Theme.HybridApp
        End Function
    End Class
    Public Class AccordionDemoModulesAccessor
        Inherits DemoModulesAccessor(Of AccordionDemoModule)

        Public Sub New(ByVal fixture As BaseDemoTestingFixture)
            MyBase.New(fixture)
        End Sub
    End Class
    Public MustInherit Class BaseAccordionDemoTestingFixture
        Inherits BaseDemoTestingFixture

        Private privateModuleAccessor As AccordionDemoModulesAccessor
        Protected Property ModuleAccessor() As AccordionDemoModulesAccessor
            Get
                Return privateModuleAccessor
            End Get
            Private Set(ByVal value As AccordionDemoModulesAccessor)
                privateModuleAccessor = value
            End Set
        End Property
        Public Sub New()
            ModuleAccessor = GetModulesAccessor()
        End Sub
        Protected MustOverride Function GetModulesAccessor() As AccordionDemoModulesAccessor
    End Class
End Namespace
